aten/src/ATen/native/PixelShuffle.cpp

```diff
@@ -4,31 +4,45 @@
 #include <c10/util/Exception.h>
 
 #include <algorithm>
+#include <numeric>
 #include <vector>
 
 namespace at {
 namespace native {
 
 Tensor pixel_shuffle(const Tensor& self, int64_t upscale_factor) {
-  AT_ASSERTM(self.dim() == 4,
-             "pixel_shuffle expects 4D input, but got input with sizes ",self.sizes());
-  int64_t b = self.size(0);
-  int64_t c = self.size(1);
-  int64_t h = self.size(2);
-  int64_t w = self.size(3);
+  TORCH_CHECK(self.dim() >= 3,
+             "pixel_shuffle expects at least 3D input, but got input with sizes ", self.sizes());
+  // Format: (B1, ..., Bn), C, H, W
+  int64_t c = self.size(-3);
+  int64_t h = self.size(-2);
+  int64_t w = self.size(-1);
   int64_t upscale_factor_squared = upscale_factor * upscale_factor;
-  AT_ASSERTM(c % upscale_factor_squared == 0,
+  TORCH_CHECK(c % upscale_factor_squared == 0,
              "pixel_shuffle expects input channel to be divisible by square of "
              "upscale_factor, but got input with sizes ", self.sizes(),
```

Comment:
the construction here is a little awkward:

"..., but got input with sizes X, upscale_factor=Y, and self.size(-3)=Z is not divisible by A"

Doubly so since the name "self" is not visible to the user in this context.

One fix would be to simplify the message to:

"...but input.size(-3)=X is not divisible by Y".